gcloud-java-storage/src/main/java/com/google/gcloud/storage/ListResult.java

```diff
@@ -16,67 +16,28 @@
 
 package com.google.gcloud.storage;
 
-
-import java.io.Serializable;
-import java.util.Collections;
 import java.util.Iterator;
-import java.util.Objects;
 
 /**
- * Google Cloud storage list result.
+ * Interface for Google Cloud storage list result.
  */
-public final class ListResult<T extends Serializable> implements Iterable<T>, Serializable {
-
-  private static final long serialVersionUID = -6937287874908527950L;
-
-  private final String cursor;
-  private final Iterable<T> results;
-  private final NextPageFetcher<T> pageFetcher;
-
-  public interface NextPageFetcher<T extends Serializable> extends Serializable {
-    ListResult<T> nextPage();
-  }
-
-  public ListResult(NextPageFetcher<T> pageFetcher, String cursor, Iterable<T> results) {
-    this.pageFetcher = pageFetcher;
-    this.cursor = cursor;
-    this.results = results;
-  }
+public interface ListResult<T> extends Iterable<T> {
 
   /**
    * Returns the cursor for the nextPage or {@code null} if no more results.
+   * 
+   * @return the string cursor for next page
```

Comment:
I think the preferred style in this case (simple "Returns XXX" statement) would be without the extra "@return".
This [link](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment) does directly state that, but is related.